File: examples/mops/mops.cs
/*
Copyright (C) 2001-2003 The Perl Foundation.  All rights reserved.
$Id$

=head1 NAME

examples/mops/mops.c - Calculate M ops/s

=head1 SYNOPSIS

I<Anyone know how to run this?>

=head1 DESCRIPTION

A C# implementation of the F<examples/mops/mops.pasm> example
program, for speed comparisons.

Calculates a value for M ops/s (million operations per second) using
integer arithmetic.

Prints out:

=over 4

=item * the number of look iterations, 

=item * the estimated number of ops performed,

=item * the elapsed time, and 

=item * the number of M op/s.

=back

=cut

*/

// created on 03/03/2002 at 15:12
using System;

class App {
	public static int Main(String[] args) {
		int i1, i5;
		// int i2 = 0;
		int i3 = 1;
		int i4 = 100000000;
		DateTime start, end;
		double n1, n2;
		Console.WriteLine("Iterations:    {0}", i4);

//  IL_0038:  ldloc.3
//  IL_0039:  ldloc.2
//  IL_003a:  sub
//  IL_003b:  stloc.3
//  IL_003c:  ldloc.3
//  IL_003d:  ldc.i4.0
//  IL_003e:  conv.i8
//  IL_003f:  bgt.s      IL_0038
//
// Insane, isn't it?
		i1 = 8;
		i5 = i4 * i1;
		Console.WriteLine("Estimated ops: {0}", i5);
		start = DateTime.Now;
		REDO:
			i4 = i4 - i3;
		if (i4 > 0)
			goto REDO;
		end = DateTime.Now;
		n2 = (end-start).TotalMilliseconds;
		n2 /= 1000; // Milliseconds to seconds
		Console.WriteLine("Elapsed Time:  {0}", n2);
		n1 = i5;
		n1 /= n2;
		n2 = 1000000.0;
		n1 /= n2;
		Console.WriteLine("M op/s:        {0}", n1);
		return 0;
	}
}

/*

=head1 SEE ALSO

F<examples/benchmarks/mops.pasm>,
F<examples/mops/mops.c>,
F<examples/mops/mops.il>,
F<examples/mops/mops.p6>,
F<examples/mops/mops.pl>,
F<examples/mops/mops.ps>,
F<examples/mops/mops.py>,
F<examples/mops/mops.rb>,
F<examples/mops/mops.scheme>.

=cut

*/

File: examples/mops/mops.py
"""
# Copyright (C) 2001-2003 The Perl Foundation.  All rights reserved.
# $Id$

=head1 NAME

examples/mops/mops.c - Calculate M ops/s

=head1 SYNOPSIS

    % python examples/mops/mops.py

=head1 DESCRIPTION

A Python implementation of the F<examples/mops/mops.pasm> example
program, for speed comparisons.

Calculates a value for M ops/s (million operations per second) using
integer arithmetic.

Prints out:

=over 4

=item * the number of look iterations, 

=item * the estimated number of ops performed,

=item * the elapsed time, and 

=item * the number of M op/s.

=back

=cut

"""

import sys
import time

I2 = 0                       # set    I2, 0
I3 = 1                       # set    I3, 1
I4 = 10000000                # set    I4, 10000000
                             #
print "Iterations:   ", I4   # print  "Iterations:    "
                             # print  I4
                             # print  "\n"
                             #
I1 = 2                       # set    I1, 2
I5 = I4 * I1                 # mul    I5, I4, I1
                             #
print "Estimated ops:", I5   # print  "Estimated ops: "
                             # print  I5
                             # print  "\n"
                             #
N1 = time.time()             # time N1
                             #
while I4:                    # REDO:
  I4 = I4 - I3               # sub    I4, I4, I3
                             # if     I4, REDO
                             #
                             # DONE:
N5 = time.time()             # time   N5
                             #
N2 = N5 - N1                 # sub    N2, N5, N1
                             #
print "Elapsed time: ", N2   # print  "Elapsed time:  "
                             # print  N2
                             # print  "\n"
                             #
N1 = I5                      # iton   N1, I5
N1 = N1 / N2                 # div    N1, N1, N2
N2 = 1000000.0               # set    N2, 1000000.0
N1 = N1 / N2                 # div    N1, N1, N2
                             #
print "M op/s:       ", N1   # print  "M op/s:        "
                             # print  N1
                             # print  "\n"
                             #
sys.exit(0)                  # end

"""

=head1 SEE ALSO

F<examples/assembly/mops.pasm>,
F<examples/mops/mops.c>,
F<examples/mops/mops.cs>,
F<examples/mops/mops.il>,
F<examples/mops/mops.p6>,
F<examples/mops/mops.pl>,
F<examples/mops/mops.ps>,
F<examples/mops/mops.rb>,
F<examples/mops/mops.scheme>.

=cut

"""